webviews/wkwebview/wkwebview/LinkViewController.swift

```diff
@@ -2,138 +2,185 @@
-    func generateLinkInitializationURL() -> String {
-        // Create a new link_token via the /link/token/create endpoint. You will be
-        // able to configure this link_token to control Link behavior. To learn more
-        // about how to create a link_token, visit https://plaid.com/docs/#create-link-token.
-        //
-        // After creating a link_token, replace <#GENERATED_LINK_TOKEN#> with it below.
-        let config = [
-            "token": "<#GENERATED_LINK_TOKEN#>"
-            "isMobile": "true",
-            "isWebview": "true",
-        ]
-
-        // Build a dictionary with the Link configuration options
-        // See the Link docs (https://plaid.com/docs/quickstart) for full documentation.
-        var components = URLComponents()
-        components.scheme = "https"
-        components.host = "cdn.plaid.com"
-        components.path = "/link/v2/stable/link.html"
-        components.queryItems = config.map { URLQueryItem(name: $0, value: $1) }
-        return components.string!
-    }
+    // MARK: WKNavigationDelegate
 
     func webView(_ webView: WKWebView,
                  decidePolicyFor navigationAction: WKNavigationAction,
-                 decisionHandler: @escaping ((WKNavigationActionPolicy) -> Void)) {
+                 decisionHandler: @escaping ((WKNavigationActionPolicy) -> Void)
+    ) {
+        guard let url = navigationAction.request.url else {
+            decisionHandler(.allow)
+            return
+        }
 
         let linkScheme = "plaidlink";
-        let actionScheme = navigationAction.request.url?.scheme;
-        let actionType = navigationAction.request.url?.host;
-        let queryParams = getUrlParams(url: navigationAction.request.url!)
+        let actionScheme = url.scheme;
+        let actionType = url.host ?? "";
+        let queryParams = getUrlParams(url: url)
 
-        if (actionScheme == linkScheme) {
+        if actionScheme == linkScheme {
             switch actionType {
-
-            case "connected"?:
+            case "connected":
                 // Close the webview
                 _ = self.navigationController?.popViewController(animated: true)
 
                 // Parse data passed from Link into a dictionary
                 // This includes the public_token as well as account and institution metadata
-                print("Public Token: \(queryParams["public_token"])");
-                print("Account ID: \(queryParams["account_id"])");
-                print("Institution type: \(queryParams["institution_type"])");
-                print("Institution name: \(queryParams["institution_name"])");
+                print("Public Token: \(queryParams["public_token"] ?? "")");
+                print("Account ID: \(queryParams["account_id"] ?? "")");
+                print("Institution type: \(queryParams["institution_type"] ?? "")");
+                print("Institution name: \(queryParams["institution_name"] ?? "")");
```

Comment:
🖼️ This could possibly be more swifty:
```suggestion
                print(
                    """
                    Public Token: \(queryParams["public_token"] ?? "")
                    Account ID: \(queryParams["account_id"] ?? "")
                    Institution type: \(queryParams["institution_type"] ?? "")
                    Institution name: \(queryParams["institution_name"] ?? "")
                    """
                )
```